joern-cli/frontends/swiftsrc2cpg/src/main/scala/io/joern/swiftsrc2cpg/astcreation/AstForStmtSyntaxCreator.scala

```diff
@@ -73,7 +77,466 @@ trait AstForStmtSyntaxCreator(implicit withSchemaValidation: ValidationMode) {
     Ast(controlStructureNode(node, ControlStructureTypes.CONTINUE, code(node)))
   }
 
-  private def astForForStmtSyntax(node: ForStmtSyntax): Ast     = notHandledYet(node)
+  private def extractLoopVariableNodeInfo(binding: ValueBindingPatternSyntax): Option[PatternSyntax] = {
+    binding.pattern match {
+      case id: IdentifierPatternSyntax                        => Some(id)
+      case expr: ExpressionPatternSyntax                      => Some(expr)
+      case tuple: TuplePatternSyntax                          => Some(tuple)
+      case _: WildcardPatternSyntax | _: MissingPatternSyntax => Some(binding.pattern)
+      case _                                                  => None
+    }
+  }
+
+  private def astForForStmtBody(node: ForStmtSyntax): Ast = {
+    node.whereClause match {
+      case Some(whereClause: WhereClauseSyntax) =>
+        val ifNode  = controlStructureNode(whereClause.condition, ControlStructureTypes.IF, code(whereClause.condition))
+        val testAst = astForNode(whereClause)
+        val consequentAst = astForNode(node.body)
+        setOrderExplicitly(testAst, 1)
+        setOrderExplicitly(consequentAst, 2)
+        Ast(ifNode)
+          .withChild(testAst)
+          .withConditionEdge(ifNode, testAst.nodes.head)
+          .withChild(consequentAst)
+      case None => astForNode(node.body)
+    }
+  }
+
+  private def astForForStmtSyntaxWithWildcard(node: ForStmtSyntax): Ast = {
+    val initAsts = Seq(astForNodeWithFunctionReference(node.sequence))
+    val bodyAst  = astForForStmtBody(node)
+    val forNode  = controlStructureNode(node, ControlStructureTypes.FOR, code(node))
+    forAst(forNode, Nil, initAsts, Nil, Nil, bodyAst)
+  }
+
+  /** De-sugaring from:
+    *
+    * for i in arr { body }
+    *
+    * to:
+    *
+    * { var _iterator = <operator>.iterator(arr); var _result; var i; while (!(_result = _iterator.next()).done) { i =
+    * _result.value; body } }
+    */
+  private def astForForStmtSyntaxWithIdentifier(node: ForStmtSyntax, id: IdentifierPatternSyntax): Ast = {
+    // surrounding block:
+    val blockNode_ = blockNode(node)
+    scope.pushNewBlockScope(blockNode_)
+    localAstParentStack.push(blockNode_)
+
+    val collection     = node.sequence
+    val collectionName = code(collection)
+
+    // _iterator assignment:
+    val iteratorName      = generateUnusedVariableName(usedVariableNames, "_iterator")
+    val iteratorLocalNode = localNode(node, iteratorName, iteratorName, Defines.Any).order(0)
+    val iteratorNode      = identifierNode(node, iteratorName)
+    diffGraph.addEdge(localAstParentStack.head, iteratorLocalNode, EdgeTypes.AST)
+    scope.addVariableReference(iteratorName, iteratorNode)
+
+    val iteratorCall =
+      // TODO: add operator to schema
+      callNode(node, s"<operator>.iterator($collectionName)", "<operator>.iterator", DispatchTypes.STATIC_DISPATCH)
```

Comment:
Note for the future: If we're adding support for an `iterator` operator, then I think it would make sense to use that across frontends through a common `foreach` lowering in `x2cpg`. This would be a fairly significant representation change for existing frontends, however.

Reply:
Yes, true. Currently, its done this way in js2cpg, jssrc2cpg, and afaik pysrc2cpg.
This should be refactored at some point.